src/briefcase/console.py

```diff
@@ -137,6 +137,33 @@ def __init__(self, printer=Printer(), verbosity=1):
         self.stacktraces = []
         # functions to run for additional logging if creating a logfile
         self.log_file_extras = []
+        # The current context for the log
+        self._context = ""
+
+    @contextmanager
+    def context(self, context):
+        """Wrap a collection of output in a logging context.
+
+        A logging context is a prefix on every logging line. It is used when a
+        set of commands (and output) is being run in a very specific way that
+        needs to be highlighted, such as running a command in a Docker
+        container.
+
+        :param context: The name of the context to enter. This *must* be
+            simple text, with no markup or other special characters.
+        """
+        try:
+            self.info()
+            self.info(f"Entering {context} context...")
+            old_context = self._context
+            self._context = f"{context}| "
+            self.info("-" * (72 - len(context)))
+            yield
+        finally:
+            self.info("-" * (72 - len(context)))
+            self._context = old_context
+            self.info(f"Leaving {context} context.")
+            self.info()
```

Comment:
nitpick...throwing this out there.

The vertical spacing here is bugging me a bit because we've usually reserved vertical space for separating distinct tasks.

```
[helloworld] Building application...
Build bootstrap binary...

Entering Docker context...
Docker| ------------------------------------------------------------------
Docker| make: Entering directory '/app/ubuntu/jammy/helloworld/bootstrap'
Docker| mkdir -p ../helloworld-0.0.1/usr/bin
Docker| cp helloworld ../helloworld-0.0.1/usr/bin
Docker| make: Leaving directory '/app/ubuntu/jammy/helloworld/bootstrap'
Docker| ------------------------------------------------------------------
Leaving Docker context.

Building bootstrap binary... done
Installing license... done
```

Here, though, we're actually doing the task we just mentioned. So, i think it might be improved by removing `Entering/Leaving` lines as well as the vertical space.

```
[helloworld] Building application...
Build bootstrap binary...
Docker| ------------------------------------------------------------------
Docker| make: Entering directory '/app/ubuntu/jammy/helloworld/bootstrap'
Docker| mkdir -p ../helloworld-0.0.1/usr/bin
Docker| cp helloworld ../helloworld-0.0.1/usr/bin
Docker| make: Leaving directory '/app/ubuntu/jammy/helloworld/bootstrap'
Docker| ------------------------------------------------------------------
Building bootstrap binary... done
Installing license... done
```

not especially married to either format though...

Reply:
I can see your point, but I think moving into Docker is enough of an "implicit context switch" that an extra space makes sense.